### PR TITLE
Do not depend on noex::Table::AsyncApply().

### PR DESCRIPTION
**Pull request by rsimion**
Part of the work for #2097 
Fourth request in a series of pull requests to move functions from noex:;Table to Table.

---
This change is [Reviewable](https://reviewable.io/reviews/googleapis/google-cloud-cpp/2292)

### CODE REVIEW

File: google/cloud/bigtable/table.cc [hunk 1]
```diff
@@ -74,24 +76,38 @@ Status Table::Apply(SingleRowMutation mut) {
...
+             internal::ConstantIdempotencyPolicy(is_idempotent),
+             clone_metadata_update_policy(),
+             [client](grpc::ClientContext* context,
+                      google::bigtable::v2::MutateRowRequest const& request,
+                      grpc::CompletionQueue* cq) {
+               return client->AsyncMutateRow(context, request, cq);
+             },
+             std::move(request), cq)
+      .then([](future<StatusOr<google::bigtable::v2::MutateRowResponse>> r) {
+        return r.get().status();
```
Comment (remyabel) [inline google/cloud/bigtable/table.cc]: Should you check `if (r.get())` before getting `status()`?

Comment (coryan) [reply google/cloud/bigtable/table.cc]: I do not think we need to do that:

- If `r.get().ok()` is false, then we want to return `.status()`.
- If `r.get().ok()` is true, then `.status() == Status()` and that is exactly what we want to return.
- Our code is not supposed to throw, so the future should not contain an exception, but if it did, `r.get()` throws and the exception is captured in the future returned by `.then()`.

File: google/cloud/bigtable/table.cc [hunk 2]
```diff
@@ -20,6 +20,8 @@
 #include <thread>
 #include <type_traits>
 
+#include "google/cloud/bigtable/internal/async_retry_unary_rpc.h"
```
Comment (coryan) [inline google/cloud/bigtable/table.cc]: Please move this `#include` to the block of includes, note that the block will be sorted by `clang-format`.

File: google/cloud/bigtable/table.cc [hunk 3]
```diff
@@ -74,24 +76,38 @@ Status Table::Apply(SingleRowMutation mut) {
...
+  internal::SetCommonTableOperationRequest<
+      google::bigtable::v2::MutateRowRequest>(
+      request, impl_.app_profile_id_.get(), impl_.table_name_.get());
+  mut.MoveTo(request);
+  auto context = google::cloud::internal::make_unique<grpc::ClientContext>();
+
+  // Determine if all the mutations are idempotent. The idempotency of the
+  // mutations won't change as the retry loop executes, so we can just compute
+  // it once and use a constant value for the loop.
+
```
Comment (coryan) [inline google/cloud/bigtable/table.cc]: nit: remove this blank line please.

**Pull request merged**